src/main/scala/org/constellation/storage/TransactionService.scala

```diff
@@ -99,6 +100,24 @@ class TransactionService[F[_]: Concurrent](dao: DAO)
       .flatTap(txs => Sync[F].delay(logger.debug(s"Pulling txs=${txs.size} for consensus with minCount: $minCount")))
       .flatMap(_.traverse(tx => withLock("inConsensusPut", inConsensus.put(tx.transaction.hash, tx))))
 
+  def pullForConsensusIfEmptyCreateDummyTransaction(
+    minCount: Int,
+    roundId: String = "roundId"
+  ): F[List[TransactionCacheData]] =
+    count(status = TransactionStatus.Pending).flatMap {
+      case 0L => LiftIO[F].liftIO(dao.transactionService.dummyTransaction())
```

Comment:
This code is in `TransactionService` so just use it directly, not through the `dao`